src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -176,6 +176,21 @@ public ODataMessageQuotas MessageQuotas
         /// </summary>
         public bool ReadUntypedAsString { get; set; }
 
+        /// <summary>
+        /// Func to evaluate whether a property should be read as a stream.  Note that IEdmProperty may be null when reading
+        /// within a collection
+        /// </summary>
+        /// <Remarks>
+        /// Function takes:
+        /// * Primitive type of the value being read, or null if unknown
+        /// * Whether the value being read is a collection
+        /// * The name of the property being read (null for values within a collection)
+        /// * The property being read (null for dynamic property or value within a collection)
+        /// Function returns:
+        /// * True, to have the value streamed, otherwise false
+        /// </Remarks>
+        public Func<IEdmPrimitiveType, bool, string, IEdmProperty, bool> ReadAsStream { get; set; }

```

Comment:
>ReadAsStream [](start = 73, length = 12)

Maybe name as "ReadAsStreamFunc"

Reply:
Fixed.

---
In reply to: [258651417](https://github.com/OData/odata.net/pull/1204#discussion_r258651417) [](ancestors = 258651417)